plugins/net_plugin/net_plugin.cpp

```diff
@@ -2489,7 +2488,18 @@ namespace eosio {
             boost::asio::bind_executor( strand,
               [conn = shared_from_this(), socket=socket]( boost::system::error_code ec, std::size_t bytes_transferred ) {
                // may have closed connection and cleared pending_message_buffer
-               if( !conn->socket_is_open() || socket != conn->socket ) return;
+               if (!conn->socket_is_open() && conn->socket_open) { // if socket_open then close not called
+                  peer_dlog( conn, "async_read socket not open, closing");
+                  conn->close();
```

Comment:
This will set a 100ms timer to attempt a reconnection. Is this what we want?

Reply:
Yes.